src/htcp.cc

```diff
@@ -923,7 +923,8 @@ htcpSpecifier::checkHit()
         return;
     }
 
-    if (!checkHitRequest->header.parse(req_hdrs, reqHdrsSz)) {
+    Http::ContentLengthInterpreter interpreter;
+    if (!checkHitRequest->header.parse(req_hdrs, reqHdrsSz, interpreter)) {
```

Comment:
I would add an HttpRequest::parseHeader(buffer, size) method that can be used here and in similar contexts. The method will declare ContentLengthInterpreter, of course. It is wrong to reach into an object to manipulate its parts that may depend on other object fields (e.g., request method) and adding this HttpRequest method would be a step in the right direction even without the need to hide ContentLengthInterpreter.

There is already a Message::parseHeader(Http::One::Parser&) method. I would try giving it an clen interpreter parameter and make the method protected. Message kids will define public parseHeader(Http::One::Parser&) and pass the right interpreter parameter to the protected parent method, avoiding new virtual calls.

Callers should have no problem deciding which parseHeader() method to call because those who poses a Http::One::Parser object cannot easily get to (buf, size) and vice versa.

As the existing Message::parseHeader() TODO says, the whole approach to placing parses into objects they are supposed to create is wrong, but fixing that is outside this small PR scope, and the above changes do not make the situation worse. Eventually, the One::Parser will do all the parsing and will have a clen interpreter data member to deal with Content-Length header fields.